source/features/show-local-user-time.tsx

```diff
@@ -0,0 +1,148 @@
+/* eslint-disable no-await-in-loop */
+
+import mem from 'mem';
+import React from 'dom-chef';
+import select from 'select-dom';
+import features from '../libs/features';
+import observeEl from '../libs/simplified-element-observer';
+import {clock} from '../libs/icons';
+import * as api from '../libs/api';
+
+interface Commit {
+	url: string;
+}
+
+interface Event {
+	type: string;
+	payload: {
+		commits: Commit[];
+	};
+	actor: {
+		id: number;
+	};
+}
+
+const normalizeApiUrl = (url: string): string => {
+	const {pathname} = new URL(url);
+	return pathname.replace(/^\/(api\/v3\/)?/, '');
+};
+
+async function loadLastCommit(pathname: string): Promise<Commit|undefined> {
+	const response = await api.v3(pathname);
+
+	// NOTE: We iterate in series here to reduce requests
+	// The first commit should already work in most cases
+	const events = response.filter((event: Event) => event.type === 'PushEvent') as Event[];
+	for (const event of events) {
+		// NOTE: We want to iterate the commits in reverse to start with the latest commitjust
+		for (const commit of event.payload.commits.reverse()) {
+			const response = await api.v3(normalizeApiUrl(commit.url), {ignoreHTTPStatus: true});
+			if (!response.ok) {
+				// NOTE: Ignore 404 errors and check the next commit
+				if (response.httpStatus === 404) {
+					continue;
+				}
+
+				return;
+			}
```

Comment:
Yes 😄  
I ran into this on the [contributors page](https://github.com/sindresorhus/refined-github/graphs/contributors), but I don't have the commit anymore. 🙈 
My guess would that this could happen through a force push. 🤷‍♂

Reply:
Force pushes don’t deleted commits, they’re still on GitHub even if they’re no longer directly accessible. 

The only situation where this could happen is if a PushEvent is still listed in the events even if the repo is deleted. My assumption is that after a repo deletion, all of its events are dropped from the list. If my assumption is wrong, then yes 404s are possible in this case.

Reply:
I dont think it is.

Reply:
Indeed. Verified

<img width="785" alt="" src="https://user-images.githubusercontent.com/1402241/69683159-5a327480-10e6-11ea-8428-8222dadde2bf.png">

```yml
$ http 'https://api.github.com/repos/fregante/temp/commits/2b9bc9d637176d6522659c16c15aaa43a41ad047' Accept:application/vnd.github.v3.patch

HTTP/1.1 404 Not Found
Access-Control-Allow-Origin: *
Access-Control-Expose-Headers: ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type
Content-Encoding: gzip
Content-Security-Policy: default-src 'none'
Content-Type: application/json; charset=utf-8
Date: Wed, 27 Nov 2019 00:19:19 GMT
Referrer-Policy: origin-when-cross-origin, strict-origin-when-cross-origin
Server: GitHub.com
Status: 404 Not Found
Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
Transfer-Encoding: chunked
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-GitHub-Media-Type: github.v3; param=patch
X-GitHub-Request-Id: CC8F:04BD:2253F8:2C58D2:5DDDC107
X-RateLimit-Limit: 60
X-RateLimit-Remaining: 54
X-RateLimit-Reset: 1574817518
X-XSS-Protection: 1; mode=block

{
    "documentation_url": "https://developer.github.com/v3/repos/commits/#get-a-single-commit",
    "message": "Not Found"
}
```

Reply:
I think only `response.httpStatus === 404` needs to be there though

Reply:
Okay I handled this now :+1:

I also exposed the `getError` function from `libs/api` to throw an error when `response.httpStatus !== 404`.

I also assume that any other commit of that `PushEvent` would be gone and `break` out of the loop instead of `continue`.